autopush/tests/certs/makecerts.py

```diff
@@ -0,0 +1,71 @@
+# encoding: utf-8
+
+import uuid
+from inspect import getsource
+
+from OpenSSL.crypto import (
+    FILETYPE_PEM, PKey, TYPE_RSA, X509, X509Extension,
+    dump_certificate, dump_privatekey)
+
+SHABREAK = 51
+
+
+def make_cert(filename, cacert=None, cakey=None):
+    key = PKey()
+    key.generate_key(TYPE_RSA, 2048)
+
+    cert = X509()
+    subject = cert.get_subject()
+
+    subject.C = b"TR"
+    subject.ST = b"Çorum"
+    subject.L = b"Başmakçı"
+    subject.CN = b"localhost"
+    subject.O = b"Mozilla Test"
+    subject.OU = b"Autopush Test %s" % filename
+    subject.emailAddress = b"otto.push@example.com"
+    subjectAltName = X509Extension(b'subjectAltName', False, b'DNS:localhost')
+    cert.add_extensions([subjectAltName])
+
+    cert.set_serial_number(uuid.uuid4().int)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(60 * 60 * 24 * 365 * 100)
+
+    cert.set_pubkey(key)
+
+    if not cacert:
+        # self sign
+        cacert = cert
+        cakey = key
+    cert.set_issuer(cacert.get_subject())
+    cert.sign(cakey, 'sha1')
+
+    with open(filename, 'wb') as fp:
+        fp.write(dump_privatekey(FILETYPE_PEM, key))
+        fp.write(dump_certificate(FILETYPE_PEM, cert))
+    return cert, key
+
+
+def save_sha256(sha256):
+    import __main__
+    source = getsource(__main__)
+    source = source.splitlines(True)[:-2]
+    with open('makecerts.py', 'wb') as fp:
```

Comment:
uhm, eek? Any reason to not store this as something potentially executable? Say, as XML or plain text you read in with constraints?